### PR TITLE
- Add introductory_price_duration to product price map

### PR DESCRIPTION
**Pull request by mshmoustafa**
Big rewrite (sorry) but this includes the following changes:

## README

- Moves the sample files to platform folders (e.g. iOS sample files moved to the iOS folder, etc.)
- Adds a title to the README
- Adds a table summarizing which fields to include for each platform
- Explains the fields of a product price map
- Adds descriptions of Android and Stripe fields for consistency
- Adds a note that Stripe subscription objects contain only the current subscription status (somewhat similar to Android behavior.)
- Adds a section explaining how to do an import with anonymous user IDs.
- Lots of minor formatting and tweaks (moving notes to callouts, bolding important points, etc.)

## ios_product_price_map_sample.csv 

- Adds `introductory_price_duration` column

## How to review

I flagged all of the new content below in my review, so focusing on those would be great. Aside from that, a quick look over the [Markdown preview](https://github.com/RevenueCat-Samples/import-csv-samples/tree/sharif/add-introductory-price-duration) to make sure everything's in the right place would also be great.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,41 +1,112 @@
+# Migrating receipts to RevenueCat via bulk import
```
Comment (mshmoustafa) [inline README.md:1]: This guide is turning into documentation so a title might help discoverability and scannability

File: README.md [hunk 2]
```diff
@@ -1,41 +1,112 @@
...
+
+`introductory_price` The introductory offer price. This column is required if you offer an introductory period that is not a free trial.
+
+`currency` The currency of the price. If you don't have prices and currencies, you need to provide a product price map file.
+
+> If you don't have product IDs, prices or currencies with the receipts we can take a [separate file](#product-price-map-ios-only) with a mapping of product_id, price, currency as well. If there were any known price changes, those should also be included here.
+
+***Note: The latest_receipt_info is not supported as it's only a subset of the entire receipt.** The entire iOS receipt file must be used for imports.
+
+#### Product Price Map (iOS Only)
```
Comment (mshmoustafa) [inline README.md:41]: This whole product price map section is new, 👀 appreciated!

File: README.md [hunk 3]
```diff
@@ -1,41 +1,112 @@
...
+> 💡 Check out the [sample file](iOS/ios_product_price_map_sample.csv) for a complete example of a product price map CSV file for iOS.
 
-`token`
+`app_user_id` The user identifier to associate with the subscription.
 
-Important: We need to know your API quota with Google and approximately how many remaining requests per day you have so we don't exceed your quota.
+`subscription_token` (e.g. `sub_xxxxxx`) The subscription token.*
 
-*Note that Android receipts that expired more than 90 days ago can't be imported, and only the current status can be retrieved from each Android receipt. This means that Android charts for historical data won't be accurate.
+***Note that Stripe subscription objects contain only the current status of the subscription.** This means that Stripe charts for historical data won't be accurate.
```
Comment (mshmoustafa) [inline README.md:105]: This callout is also new to bring attention to the fact that Stripe doesn't store complete subscription history in the token

File: README.md [hunk 4]
```diff
@@ -1,41 +1,112 @@
...
 
-## Stripe:
+## Receipt imports for apps without user authentication systems
```
Comment (mshmoustafa) [inline README.md:107]: This section is completely new, 👀 appreciated!

File: iOS/ios_product_price_map_sample.csv [hunk 5]
```diff
@@ -0,0 +1,4 @@
+product_id,price,currency,introductory_price,date,duration,introductory_price_duration
```
Comment (mshmoustafa) [inline iOS/ios_product_price_map_sample.csv:1]: Added the optional `introductory_price_duration` column to keep this sample complete

File: README.md (hunk 2 above)
Comment (rkotzy) [reply README.md:41]: Do you think we should add an 🚧  icon or something next to some of the iOS fields that are "recommended" but not required? We could call out in the description what the implications are if this field is not provided. I know a lot of developers don't store `price` for example and need to do the Product Price Map - we could add that to the description of the `price` attribute ("If this column is not provided, a product price map is required and USD price will be assumed for all transactions")

File: README.md (hunk 3 above)
Comment (rkotzy) [reply README.md:105]: Is the Google purchase token validation period 60 days or 90 days? I told TuneIn today it was 60 days 😅

Doesn't seem clear why a product price map would be required on Android since price isn't a required field in the receipt?

File: README.md (hunk 4 above)
Comment (rkotzy) [reply README.md:107]: We could probably do the anonymous ID generation on our side at the time of import which could simplify things? May also be worth calling out that doing an import with anonymous IDs will not automatically unlock the subscription for those customers and they'll also need to do a client side migration, but they'll get all the analytics right away.

File: README.md (hunk 2 above)
Comment (mshmoustafa) [reply README.md:41]: I originally had something like that but then removed it because I wasn't sure if it would be too wordy. I'll add it back in!

File: README.md (hunk 3 above)
Comment (mshmoustafa) [reply README.md:105]: [60 days is correct](https://developer.android.com/google/play/billing/subscriptions#lifecycle), that must be an old mistake. Although if you have a grace period it might "extend" it to 90 days, maybe that's where that came from?

image

And lol, copy-paste error, neither Android nor Stripe require a product price map

File: README.md (hunk 4 above)
Comment (mshmoustafa) [reply README.md:107]: Totally would be good to generate anonymous users on our end automatically, just requires engineering work to build that in. And good point about needing to do the client side migration either way, I'll add that in.

File: README.md (hunk 2 above)
Comment (mshmoustafa) [reply README.md:41]: Changes [here](https://github.com/RevenueCat-Samples/import-csv-samples/pull/4/commits/75e2ee97140a4e9e6593cf27092eaf49be5c0951) and [here](https://github.com/RevenueCat-Samples/import-csv-samples/pull/4/commits/143ac4e5f6b46d26836b1bda2e3089b0d7d15080)

File: README.md (hunk 3 above)
Comment (mshmoustafa) [reply README.md:105]: 60 days fix [here](https://github.com/RevenueCat-Samples/import-csv-samples/pull/4/commits/fe3a2b034d7b27f65227477f3176e1d93b237353)

Wording fixed [here](https://github.com/RevenueCat-Samples/import-csv-samples/pull/4/commits/75e2ee97140a4e9e6593cf27092eaf49be5c0951)

File: README.md (hunk 4 above)
Comment (mshmoustafa) [reply README.md:107]: Client side migration note [here](https://github.com/RevenueCat-Samples/import-csv-samples/pull/4/commits/2dd77b72d93db582e1e553404ff9194ef995937a)

**Review by rkotzy — APPROVED** [review]
